backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/workpackage/impl/PurchaseModularContractLogsHandler.java

```diff
@@ -121,6 +126,15 @@ public BooleanWithReason doesRecordStateRequireLogCreation(@NonNull final I_C_Fl
 
 		final BPartnerId billBPartnerId = BPartnerId.ofRepoId(modularContractRecord.getBill_BPartner_ID());
 
+		final ProductPrice priceActual = Optional.of(modularContractRecord)
```

Comment:
i think I saw this code somewhere else too.
Maybe you could share one extractPriceActual static method

Reply:
added to `IFlatrateBL`